### DISCUSSION TITLE
make_trade_output

### DISCUSSION THREAD

**Discussion by crypto-chen** [uniswap-python/uniswap-python#309] [Q&A]
Could not identify the intended function with name `quoteExactOutputSingle`, positional argument(s) of type `(<class 'str'>, <class 'str'>, <class 'int'>, <class 'float'>, <class 'int'>)` and keyword argument(s) of type `{}`.
Found 1 function(s) with the name `quoteExactOutputSingle`: ['quoteExactOutputSingle(address,address,uint24,uint256,uint160)']
Function invocation failed due to no matching argument types.

**Comment by crypto-chen** [comment]
uniswap = Uniswap(address=address, private_key=private_key, version=version, provider=provider)

# Some token addresses we'll be using later in this guide
eth = Web3.toChecksumAddress("0x0000000000000000000000000000000000000000")
usdc = Web3.toChecksumAddress("0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8")
usdt = Web3.toChecksumAddress("0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9")

a = uniswap.make_trade_output(eth, usdc, 0.5*10**18)


print(a)

**Reply by crypto-chen** [reply]
web3.exceptions.ContractLogicError: execution reverted: Unexpected error

**Reply by ErikBjare** [reply]
Your `amount` needs to be an `int`.

**Reply by crypto-chen** [reply]
ValueError: {'code': -32601, 'message': 'the method eth_feeHistory does not exist/is not available', 'data': None}
This is my make for optimal blockchain transactions_ Error report generated by trade method

 eth = Web3.toChecksumAddress("0x0000000000000000000000000000000000000000")
    usdc = Web3.toChecksumAddress("0x7F5c764cBc14f9669B88837ca1490cCa17c31607")
    susd = web3.toChecksumAddress("0x8c6f28f2F1A3C87F0f938b96d27520d9751ec8d9")
    usdc_susd_amout = uniswap.get_price_output(susd, usdc, 21, fee=3000)
    print(usdc_susd_amout)

    tx_hash = uniswap.make_trade(usdc, susd, usdc_susd_amout, fee=3000)
    tx_has_sh = tx_hash.hex()
    print(tx_has_sh)
